test/state/mpt.cpp

```diff
@@ -160,57 +145,54 @@ void MPTNode::insert(const Path& path, bytes&& value)  // NOLINT(misc-no-recursi
     // in an existing branch node. Otherwise, we need to create new branch node
     // (possibly with an adjusted extended node) and transform existing nodes around it.
 
+    const auto [this_idx, insert_idx] = std::ranges::mismatch(m_path, path);
+
+    // insert_idx is always valid if requirements are fulfilled:
+    // - if m_path is not shorter than path they must have mismatched nibbles,
```

Comment:
```suggestion
    // - if m_path is not shorter than path they must have mismatched nibbles,  given the requirement that key uniqeness and inserted key not being a prefix if existing key,
```
if I understand correctly